Editor/Scripts/CodeGenerators/RGCodeGenerator.cs

```diff
@@ -365,25 +419,25 @@ private static List<RGStateAttributesInfo> SearchForBotStateAttributes()
                         {
                             if (!fieldDeclaration.Modifiers.Any(SyntaxKind.PublicKeyword))
                             {
-                                RGDebug.LogError($"Error: Field '{fieldDeclaration.Declaration.Variables.First().Identifier.ValueText}' in class '{className}' is not public.");
+                                RecordError($"Error: Field '{fieldDeclaration.Declaration.Variables.First().Identifier.ValueText}' in class '{className}' is not public.");
```

Comment:
(Non-blocking, since I'm happy to leave it to you to decide if it's worth it and I don't think it's worth a separate ticket for now)

Maybe give `RecordError`/`RecordWarning` an optional `SyntaxNode` parameter, and if it's provided, you can use `node.Span` to get the position in the file and `node.SyntaxTree.FilePath` to get the path to the file. Then you could include `[filename]:[line number]` in the log message. That would get us as close as possible to "properly" integrating as compiler diagnostics without having to do all the work required to do that (which is absolutely out of scope now).